components/ai_chat/core/browser/engine/engine_consumer_conversation_api_unittest.cc

```diff
@@ -3325,4 +3338,68 @@ TEST_F(EngineConsumerConversationAPIUnitTest,
                              });
 }
 
+TEST_F(EngineConsumerConversationAPIUnitTest,
+       GenerateAssistantResponse_WithSmartMode) {
+  base::test::TestFuture<EngineConsumer::GenerationResult> future;
+
+  // Create conversation history with smart mode entry
+  EngineConsumer::ConversationHistory conversation_history;
+  auto smart_mode_entry =
+      mojom::SmartModeEntry::New("summarize", "Please summarize the content");
+  conversation_history.push_back(mojom::ConversationTurn::New(
+      "uuid", mojom::CharacterType::HUMAN, mojom::ActionType::QUERY,
+      "What is artificial intelligence?", std::nullopt /* prompt */,
+      std::nullopt /* selected_text */, std::nullopt /* events */,
+      base::Time::Now(), std::nullopt /* edits */,
+      std::nullopt /* uploaded_files */, std::move(smart_mode_entry), false,
+      std::nullopt /* model_key */));
+
+  MockConversationAPIClient* mock_client = GetMockConversationAPIClient();
+
+  // Expect that PerformRequest is called with conversation events that include
+  // both smart mode definition message and the main user message
+  EXPECT_CALL(*mock_client, PerformRequest)
+      .WillOnce([&](std::vector<ConversationEvent> conversation,
+                    const std::string& selected_language,
+                    std::optional<base::Value::List> oai_tool_definitions,
+                    const std::optional<std::string>& preferred_tool_name,
+                    mojom::ConversationCapability conversation_capability,
+                    EngineConsumer::GenerationDataCallback data_callback,
+                    EngineConsumer::GenerationCompletedCallback callback,
+                    const std::optional<std::string>& model_name) {
+        // Should have 2 events: smart mode definition + main user message
+        ASSERT_EQ(conversation.size(), 2u);
+
+        // First event should be the smart mode definition
+        EXPECT_EQ(conversation[0].role, ConversationEventRole::kUser);
+        EXPECT_EQ(conversation[0].type, ConversationEventType::kChatMessage);
+        EXPECT_EQ(GetContentStrings(conversation[0].content)[0],
+                  "Interpret '/summarize' as 'Please summarize the content' "
+                  "when handling the request.");
+
+        // Second event should be the actual user message
+        EXPECT_EQ(conversation[1].role, ConversationEventRole::kUser);
+        EXPECT_EQ(conversation[1].type, ConversationEventType::kChatMessage);
+        EXPECT_EQ(GetContentStrings(conversation[1].content)[0],
+                  "What is artificial intelligence?");
```

Comment:
Can we change the user prompt to include /summarize for better readability?

Reply:
https://github.com/brave/brave-core/pull/31422/commits/51c269c046cf72b50da0151b747aa7b5c7a87bf1